TCC.Lib/TarCompressCrypt.cs

```diff
@@ -212,7 +253,20 @@ private static string DecompressCommand(Block block, TccOption option, ExternalD
                     string passwdCommand = PasswordCommand(option, block);
                     //openssl aes-256-cbc -d -k "test" -in crypted.tar.lz4.aes | lz4 -dc --no-sparse - | tar xf -
                     cmd = $"{ext.OpenSsl().Escape()} aes-256-cbc -d {passwdCommand} -in {block.Source}";
-                    cmd += $" | {ext.Lz4().Escape()} -dc --no-sparse - ";
+                    switch (option.Algo)
+                    {
+                        case CompressionAlgo.Lz4:
+                            cmd += $" | {ext.Lz4().Escape()} -dc --no-sparse - ";
+                            break;
+                        case CompressionAlgo.Brotli:
+                            cmd += $" | {ext.Brotli().Escape()} - -d ";
+                            break;
+                        case CompressionAlgo.Zstd:
+                            cmd += $" | {ext.Zstd().Escape()} - -d ";
+                            break;
+                        default:
+                            throw new ArgumentOutOfRangeException("Unknown PasswordMode");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use a constructor overloads that allows a more meaningful exception message to be provided. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3928)